diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index c6b36bf..1b5be53 100644
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -144,7 +144,7 @@ elif [ "${LOCALVERSION+set}" != "set" ]; then
 	# If the variable LOCALVERSION is set (including being set
 	# to an empty string), we don't want to append a plus sign.
 	scm=$(scm_version --short)
-	res="$res${scm:++}"
+	res="$res"
 fi
 # finally, add the abXXX number if BUILD_NUMBER is set
 if test -n "${BUILD_NUMBER}"; then
